packages/core/src/external/aws/lambda-logic/modify-resources.ts

```diff
@@ -0,0 +1,599 @@
+    }
+  });
+
+  if (reportType && REPORT_TYPES_BLACKLIST.includes(reportType)) {
+    return undefined;
+  }
+
+  const updRes = cloneDeep(res);
+  delete updRes.code;
+  const category = res.category
+    ?.map(cat => cat.coding?.flatMap(coding => coding.display || []))
+    .join(", ");
+
+  const status = isUselessStatus(res.status) ? undefined : (res.status as string);
+
+  return {
+    ...updRes,
+    type: reportType,
+    status,
+    category,
+    presentedForm: Array.from(uniqueData),
+  };
+}
+
+type ReferenceRange = {
+  low: string | undefined;
+  high: string | undefined;
+};
+
+export type SlimObservation = Omit<
+  Observation,
+  "category" | "reading" | "value" | "interpretation" | "referenceRange" | "status" | "code"
+> & {
+  category?: string | undefined;
+  reading?: string | undefined;
+  value?: string | undefined;
+  status?: string | undefined;
+  interpretation?: string | undefined;
+  referenceRange?: ReferenceRange[] | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimObservation(res: Observation): SlimObservation {
+  const updRes = cloneDeep(res);
+  const refRange = res.referenceRange?.map(range => ({
+    low: getQuantityString(range.low),
+    high: getQuantityString(range.high),
+  }));
+  const status = isUselessStatus(res.status) ? undefined : (res.status as string);
+  delete updRes.code;
+  delete updRes.performer;
+  delete updRes.valueCodeableConcept;
+  delete updRes.valueQuantity;
+
+  return {
+    ...updRes,
+    category: getUniqueDisplaysString(res.category),
+    status,
+    reading: getUniqueDisplaysString(res.code),
+    value: res.valueCodeableConcept
+      ? getUniqueDisplaysString(res.valueCodeableConcept)
+      : getQuantityString(res.valueQuantity),
+    interpretation: getUniqueDisplaysString(res.interpretation),
+    referenceRange: refRange,
+  };
+}
+
+export type SlimMedication = Omit<Medication, "name"> & {
+  name?: string | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimMedication(res: Medication): SlimMedication {
+  const updRes = cloneDeep(res);
+  delete updRes.code;
+  return {
+    ...updRes,
+    name: getUniqueDisplaysString(res.code),
+  };
+}
+
+export type SlimMedicationRequest = Omit<MedicationRequest, "requester"> & {
+  reference?: Record<string, Partial<SlimMedication>>;
+};
+
+function slimMedicationRequest(res: MedicationRequest): SlimMedicationRequest {
+  const updRes = cloneDeep(res);
+  delete updRes.requester;
+  return {
+    ...updRes,
+  };
+}
+
+export type SlimMedicationStatement = Omit<MedicationStatement, "dosage"> & {
+  dosages?: Dosage[] | undefined;
+  reference?: Record<string, Partial<SlimMedication>>;
+};
+
+type Dosage = {
+  dose: string | undefined;
+  route: string | undefined;
+};
+
+function slimMedicationStatement(res: MedicationStatement): SlimMedicationStatement {
+  const dosages = res.dosage?.flatMap(dosage => {
+    const dose = getQuantityString(dosage.doseAndRate?.[0]?.doseQuantity);
```

Comment:
Could try to get the best from the array instead of indexing to zero

Reply:
I looked at a whole bunch of these, and never seen more than 1 doseAndRate element in the array